controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -2,19 +2,238 @@
+    def setUp(self, connect):
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+        self.mediator.get_snapshot = Mock()
+        self.mediator.get_snapshot.return_value = None
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.request = Mock()
+        self.request.secrets = {"username": "user", "password": "pass", "management_address": "mg"}
+        self.request.parameters = {}
+        self.capacity_bytes = 10
+        self.request.name = snap_name
+        self.request.source_volume_id = "A9000:12345678"
+
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
+    def test_create_snapshot_with_empty_name(self, a_enter):
+        self._test_create_object_with_empty_name(a_enter)
+
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.detect_array_type")
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__exit__")
+    def test_create_snapshot_succeeds(self, a_exit, a_enter, array_type):
+        a_enter.return_value = self.mediator
+        context = utils.FakeContext()
+        self.mediator.create_snapshot = Mock()
+        self.mediator.create_snapshot.return_value = utils.get_mock_mediator_response_snapshot(10, "snap", "wwn",
+                                                                                               "snap_vol", "xiv")
+        self.mediator.get_volume_name = Mock()
+        self.mediator.get_volume_name.return_value = snap_vol_name
+
+        array_type.return_value = "a9k"
+        self.servicer.CreateSnapshot(self.request, context)
+        self.assertEqual(context.code, grpc.StatusCode.OK)
+        self.mediator.get_snapshot.assert_called_once_with(snap_name)
+        self.mediator.create_snapshot.assert_called_once_with(snap_name, snap_vol_name)
+
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
+    def test_create_snapshot_with_wrong_secrets(self, a_enter):
+        self._test_create_object_with_wrong_secrets(a_enter)
+
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.detect_array_type")
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
+    def test_create_snapshot_with_array_connection_exception(self, a_enter, array_type):
+        self._test_create_object_with_array_connection_exception(a_enter)
+
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.detect_array_type")
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
+    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__exit__")

```

Comment:
please [fix patch/parameter order](https://github.com/IBM/ibm-block-csi-driver/pull/152/files/#r415303275) **everywhere** in your diff